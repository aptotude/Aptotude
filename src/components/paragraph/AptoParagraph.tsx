import React from 'react';
import classNames from 'classnames';
import './aptoParagraph.scss';
import { StandardTypes } from '../../utils/standardTypes';

export interface AptoParagraphDisplayProps extends StandardTypes {
  compact?: boolean;
}

const COMPONENT_PREFIX = 'AptoParagraph';

export class AptoParagraph extends React.PureComponent<AptoParagraphDisplayProps> {
  public static defaultProps = {
    compact: false
  };

  public render() {
    const {
      compact,
      className,
      forwardRef,
      children,
      ...rest
    } = this.props;

    const classes = classNames(
      COMPONENT_PREFIX,
      compact && `${COMPONENT_PREFIX}--compact`,
      className
    );

    return (
      <p
        {...rest}
        ref={forwardRef}
        className={classes}>
          {children}
      </p>
    );
  }
}
