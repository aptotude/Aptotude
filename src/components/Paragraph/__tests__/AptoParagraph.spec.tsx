import React from 'react';
import { AptoParagraph } from '../AptoParagraph';
import { render, cleanup } from 'react-testing-library';

describe('Paragraph Component', () => {
  afterEach(cleanup);

  it('Paragraph renders', () => {
    const { container } = render(<AptoParagraph />);

    const node = container.querySelector('p.AptoParagraph');
    expect(node!.className).toEqual('AptoParagraph');
  });

  it('Paragraph compact renders', () => {
    const { container } = render(<AptoParagraph compact />);

    const node = container.querySelector('p.AptoParagraph');
    expect(node!.className).toEqual('AptoParagraph AptoParagraph--compact');
  });
});
