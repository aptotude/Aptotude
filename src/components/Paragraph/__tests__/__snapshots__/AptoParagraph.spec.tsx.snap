// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph compact renders 1`] = `
<p
  className="AptoParagraph AptoParagraph--compact"
/>
`;

exports[`Paragraph renders 1`] = `
<p
  className="AptoParagraph"
/>
`;
